### DISCUSSION TITLE
Wifi LaserGRBL Control by PC is AWOL!

### DISCUSSION THREAD

**Discussion by GT-Mills** [arkypita/LaserGRBL#2328] [Q&A]
I have a new Two Trees TTS-55 I am _**TRYING**_ to control with LaserGRBL v7.12.0 with direct Wifi connection.  YES, i am able to connect my Win10 PC to the wifi link listed as  "Laser_44185" and upload files just fine.  

BUT!!!  there is NO WAY to CONTROL the thing via anything but a USB Com Port listed in the upper Left corner.  

I want to run the stinky smokey thing out in the garage, NOT MY OFFICE thank you very much.  So, like, what, I'm supposed to run a 100' USB cable from my office to the shop???  Drill a hole through the brick wall?  I have not tried to run it from a smartphone app yet, is that the only solution?  

This makes no sense.  It should connect and show up in my devices just like any other Wifi printer...or at least that's what I THINK it should do.  

Suggestions?  Tips?  Ideas?  Thanks.  I run a business making Yamaha aircraft conversion engines and got this setup to start cutting gaskets that are no longer available from the MFR.  NEED THIS ASAP!  Already have customers requesting them, promised to get them shipped by Friday  8/5/24.

**Comment by arkypita** [comment]
First of all, know that LaserGRBL is free software. Even if you paid money for your laser engraver, you are using free software written at an amateur level by a person (me) who dedicates his time for free.

This answer is also something you are getting for free, and for which the hasty and urgent tones of your request are completely out of place here. If you are really in a hurry, you should contact whoever sold you the machine, or whoever produced it.

LaserGRBL can be connected via WiFi as long as you know the IP address and the PORT (usually 23) of the laser.

You need to set LaserGRBL for the "Telnet" connection (menu "Grbl", "Settings", "Connection Protocol") and then enter the IP address followed by : and the port in the connection box.

![image](https://github.com/user-attachments/assets/ffecf5af-1770-475b-bea7-02751df1afed)


![image](https://github.com/user-attachments/assets/6a67ad2a-e9f2-4057-b759-9392a2f88248)

IP address depends on your machine and its network, if the machine has a display this is often shown on the display.

Please note that the fact that the machine has WiFi does not guarantee that LaserGRBL can connect to WiFi, it all depends on what type of connection and protocol they have implemented in the machine, you should ask the machine supplier for this information and instructions.

**Reply by GT-Mills** [reply]
THANKS!  I get it, it's free, OK but what is such an obvious, simple thing to a programmer familiar with this UI layout is not obvious at all to someone who's never seen the thing before.  When I first searched the settings window Icould not even find comm settings except the default Com bar showing the PC ports available in the default bar.  In fact, the default UI window is so dark, with zero contrast that the drop down arrows do not show up at all. Is there another setting to change this default view to the one you have in the screen shots provided here?  

This is the default view as installed - those drop down menu arrows are next to impossible to see unless you know what to look for:  

![Impossible to see drop menus](https://github.com/user-attachments/assets/16d06596-29b3-4e89-bdb4-95a00bd9f0af)

**Comment by Tribo-2dot0** [comment]
Assim eu faço para conectar meu TTS-55 por WiFi...
- Ligo a máquina e ela irá gerar um Hotspot (no seu caso Laser_44185).
- Conecto o PC nessa rede com a senha 12345678.
- Abro LaserGrbl e mudo o protocolo de conexão para Telnet; mudo também o Streaming Mode para Synchronous (para mim, Buffered da erro logo no início da execução).
- em IP:PORT coloco 192.168.4.1:8080

Pronto.
PS: mude a cor do software no menu Colors, escolha Red Laser ou Blue Laser.

**Comment by u2mejc** [comment]
Translating Tribo-2dot0, since I found this useful.
> Here is how I connect my TTS-55 via Wi-Fi...

> I turn on the machine, and it generates a hotspot (in your case, Laser_44185).
I connect my PC to that network using the password 12345678.
I open LaserGRBL and change the connection protocol to Telnet; I also change the Streaming Mode to Synchronous (for me, Buffered causes an error right at the start of the job).
In the IP:PORT field, I enter 192.168.4.1:8080.
That's it.
P.S.: Change the software color in the Colors menu—choose Red Laser or Blue Laser.
